### PR TITLE
feat: use npm to get/run bun

### PR DESCRIPTION
**Pull request by mistydemeo**
Instead of using platform-specific methods of fetching bun, swap to npm. This requires adding it to `devDependencies`.

I've also moved the actual `bun` invocations to npm scripts, with a pre-script for `bun install`.

### CODE REVIEW

File: axolotlsay-js/build.sh [hunk 1]
```diff
@@ -4,59 +4,36 @@ set -eu
...
+bun_args=
 
 # If we're building for cargo-dist, specify which target to build and what the
 # filename should be. cargo-dist will expect one output with a known filename.
 if [ -n "${CARGO_DIST_TARGET:-}" ]; then
     case "${CARGO_DIST_TARGET}" in
         x86_64-pc-windows-msvc)
-            args="--target bun-windows-x64 --outfile ${EXE_BASENAME}.exe"
+            bun_args="--target bun-windows-x64 --outfile ${EXE_BASENAME}.exe"
             ;;
```
Comment (Gankra) [inline axolotlsay-js/build.sh:18]: can we add the obvious-to-predict windows-arm64 target here for future proofing? also is musl supported by bun?

(I also wonder if making this setting variables and just having one compile expr at the end would be nicer for maintenance but it's kinda potato potahtoh)

**Review by Gankra — APPROVED** [review]
Much better!

File: axolotlsay-js/build.sh (hunk 1 above)
Comment (mistydemeo) [reply axolotlsay-js/build.sh:18]: As far as I can tell, musl hasn't launched official binaries on musl yet. That may come in the future. And yeah, good call on aarch64 Windows.

**Comment by mistydemeo** [comment]
I'm planning to tweak this a little further tomorrow before merging.

File: axolotlsay-js/build.sh (hunk 1 above)
Comment (mistydemeo) [reply axolotlsay-js/build.sh:18]: Just checked, and bun isn't supported on arm64 Windows yet. I'll make the switch statement recognize it but output x86_64 binaries.

**Pull request merged**